switmt_to_iso20022/utils.bal

```diff
@@ -0,0 +1,1462 @@
+       prtyIdn3.content.substring(0, 1).equalsIgnoreCaseAscii("/") {
+        return [(), ...validateAccountNumber(prtyIdn = prtyIdn3)];
+    }
+    return [(), (), ()];
+}
+
+# Concatenates the contents of `Nm` elements from one of two possible arrays into a single string.
+# The function handles cases where either one or both of the arrays might be provided.
+#
+# + name1 - An optional array of `Nm` elements that may contain the first set of name components.
+# + name2 - An optional array of `Nm` elements that may contain the second set of name components.
+# + return - Returns a single concatenated string of all name components, separated by spaces, or `null` if no valid 
+# input is provided.
+isolated function getName(swiftmt:Nm[]? name1, swiftmt:Nm[]? name2 = ()) returns string? {
+    string finalName = "";
+    swiftmt:Nm[] nameArray;
+    if name1 is swiftmt:Nm[] {
+        nameArray = name1;
+    } else if name2 is swiftmt:Nm[] {
+        nameArray = name2;
+    } else {
+        return ();
+    }
+    foreach int index in 0 ... nameArray.length() - 1 {
+        if index == nameArray.length() - 1 {
+            finalName += nameArray[index].content;
+            break;
+        } 
+        finalName = finalName + nameArray[index].content + " ";
+    }
+    return finalName;
+}
+
+# Extracts the country and town information from the provided `CntyNTw` array.
+# The country is extracted from the first two characters of the first element, 
+# and the town is extracted from the remaining part of the string if present.
+#
+# + cntyNTw - An optional array of `CntyNTw` elements that contains country and town information.
+# + return - Returns an array with two elements: the country (first two characters) and the town 
+# (remainder of the string), or `[null, null]` if the input is invalid.
+isolated function getCountryAndTown(swiftmt:CntyNTw[]? cntyNTw) returns string?[] {
+    string?[] cntyNTwArray = [];
+    if cntyNTw is swiftmt:CntyNTw[] {
+        cntyNTwArray.push(cntyNTw[0].content.substring(0, 2));
+        if cntyNTw[0].content.length() > 3 {
+            cntyNTwArray.push(cntyNTw[0].content.substring(3));
+            return cntyNTwArray;
+        }
+        cntyNTwArray.push(());
+        return cntyNTwArray;
+    }
+    return [(), ()];
+}
+
+# Validates an account number based on the regex pattern.
+#
+# + acc1 - Optional account number inputs to consider.
+# + acc2 - Optional account number inputs to consider.
+# + acc3 - Optional account number inputs to consider.
+# + prtyIdn - Optional party identifier that may contain an account number.
+# + return - Returns an array where the first element is the valid IBAN account number or `null`, 
+# and the second element is the invalid IBAN account number or `null`.
+isolated function validateAccountNumber(swiftmt:Acc? acc1 = (), swiftmt:PrtyIdn? prtyIdn = (), swiftmt:Acc? acc2 = (), swiftmt:Acc? acc3 = ()) returns string?[] {
```

Comment:
```suggestion
isolated function validateAccountNumber(swiftmt:Acc? acc1 = (), swiftmt:PrtyIdn? prtyIdn = (), swiftmt:Acc? acc2 = (), swiftmt:Acc? acc3 = ()) returns [string?, string?] {
```

Reply:
It is more convenient to use a tuple than an array since we know only two elements exist.

Reply:
fixed in `491631d`